src/server/common/model/application/validation.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @import {SectionValidationResult} from '../section/validation.js'
+ * @import {SectionModel} from '../section/section-model.js'
+ */
+
+/**
+ * @typedef {{[key:string]: SectionValidationResult}} ValidationResult
+ * @typedef {{
+ *   isValid: boolean,
+ *   result: ValidationResult
+ * }} ApplicationValidationResult
+ */
+
+/**
+ * Validates all the answers within the application and outputs an overall isValid result.
+ * @param {{[key:string]: SectionModel}} data - The data to be validated.
+ * @returns {ApplicationValidationResult} An object containing the validation result
+ */
+export const validateApplication = (data) => {
```

Comment:
This validation function is identical to the one we have for section but not sure where we should be moving it to

Reply:
BTW, I put in a refactor for the section version of this - mind to use that version for now?